assembly/src/parsers/serde/deserialization.rs

```diff
@@ -373,14 +373,56 @@ impl Deserializable for Instruction {
             OpCode::CDropW => Ok(Instruction::CDropW),
 
             // ----- input / output operations --------------------------------------------------------
-            OpCode::PushConstants => {
+            OpCode::PushU8 => Ok(Instruction::PushU8(bytes.read_u8()?)),
+            OpCode::PushU16 => Ok(Instruction::PushU16(bytes.read_u16()?)),
+            OpCode::PushU32 => Ok(Instruction::PushU32(bytes.read_u32()?)),
+            OpCode::PushFelt => Ok(Instruction::PushFelt(bytes.read_felt()?)),
+            OpCode::PushWord => {
+                let mut constants = Vec::new();
+                for _ in 0..4 {
+                    constants.push(bytes.read_felt()?);
+                }
+                Ok(Instruction::PushWord(
+                    constants.try_into().expect("Invalid constatnts length"),
+                ))
+            }
+            OpCode::PushU8List => {
                 let mut constants = Vec::new();
                 let length = bytes.read_u8()?;
+                validate_param_number(length)?;
+                for _ in 0..length {
+                    constants.push(bytes.read_u8()?);
+                }
+                Ok(Instruction::PushU8List(constants))
+            }
+            OpCode::PushU16List => {
+                let mut constants = Vec::new();
+                let length = bytes.read_u8()?;
+                validate_param_number(length)?;
+                for _ in 0..length {
+                    constants.push(bytes.read_u16()?);
+                }
+                Ok(Instruction::PushU16List(constants))
+            }
+            OpCode::PushU32List => {
+                let mut constants = Vec::new();
+                let length = bytes.read_u8()?;
+                validate_param_number(length)?;
+                for _ in 0..length {
+                    constants.push(bytes.read_u32()?);
+                }
+                Ok(Instruction::PushU32List(constants))
```

Comment:
You might be able to do this more compactly using iterator syntax.

Reply:
The problem is that `for_each` and `map` can not propagate an error, that could occur in `read_u<>()?` functions. I can try to use `expect()` instead of `?`, but I don't think that it's a good solution

Reply:
You should be able to use [try_for_each](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.try_for_each) and I think for `map` it should work fine as is. But maybe @vlopes11 has better suggestions?

Reply:
This looks like a

```rust
let length = bytes.read_u8()?;
validate_param_number(length)?;
(0..length)
    .map(|_| bytes.read_u32())
    .collect::<Result<_, _>>()
    .map(Instruction::PushU32List)
```

Reply:
The advantage of the approach above is that we won't allocate a vec with the possibility to resize it, depending on the number of iterations. Instead, the iterators will be smart enough to define the size for us (since range is an exact iterator) and we just parse the numbers directly into the target vec

Reply:
Awesome! I didn't know that we can use map like that. Thank you!